libkbfs/folder_block_ops.go

```diff
@@ -2585,14 +2575,36 @@ func (fbo *folderBlockOps) getDeferredWriteCountForTest(lState *lockState) int {
 	return len(fbo.deferredWrites)
 }
 
+func (fbo *folderBlockOps) updatePointer(kmd KeyMetadata, oldPtr BlockPointer, newPtr BlockPointer) {
+	updated := fbo.nodeCache.UpdatePointer(oldPtr.Ref(), newPtr)
+	if !updated {
+		return
+	}
+	// Prefetch the new reference, but only if it already exists in the block
```

Comment:
Sorry, I missed that `NewEmpty` was being called before on the same reference for that cache (I thought it was being done inside the prefetcher somewhere).

Hmm, that's a bummer. Please at least add a summary of our conversation here in the comments.  But we can actually tell if this is a directory block from the caller of this function, based on information in the MD update where the `op` originated.  It would get a little complicated with `resolutionOp`s though, so it might be best to add a TODO and a new JIRA issue for it for now.